### ISSUE TITLE
Maximum recursion depth exceed while chaining filters with .must

### ISSUE THREAD

**Issue by 3lnc** [elastic/elasticsearch-dsl-py#135]
Chaining two boolean filters with `must` or `must` fails due to recursion depth.

```python
from elasticsearch_dsl import F

f1 = F('missing', field='a')
f2 = F('missing', field='b')
f3 = F('missing', field='c')
f4 = F('missing', field='d')

f1 & f2 | f4 & f4
```
as chaining so calls `DslBase.__or__` -> `BoolMixin.__ror__` -> `super(BoolMixin, self).__or__`

**Comment by HonzaKral** [comment]
Thanks for the report, I cannot seem to replicate, is it possible that it was fixed as 5db919ab64adef8dfd68bedefcfe404a75eeae19 (fix for #96)

**Comment by 3lnc** [comment]
Yeah, seems so. Forget to check code from master.
